tsdb/index/tsi1/partition.go

```diff
@@ -354,6 +354,26 @@ func (p *Partition) CurrentCompactionN() int {
 	return p.currentCompactionN
 }
 
+// WaitWithTimeout will busy loop while checking CurrentCompactionN
+// If the loop goes for > duration it will return true (timedOut)
+// if it does not time out it returns false (!timedOut)
+func (p *Partition) WaitWithTimeout(duration time.Duration) bool {
+	timeout := time.NewTimer(duration)
```

Comment:
I would get rid of `timeout` and just have a ticker. In the tick `case`, first check for `p.CurrentCompactionN() == 0` then check if `time.Since(startOfMethod) > duration` where `startOfMethod` is a `time.Now()` on function entry